app/src/main/java/org/hackillinois/android/view/ProfileFragment.kt

```diff
@@ -0,0 +1,121 @@
+package org.hackillinois.android.view
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.graphics.Bitmap
+import android.graphics.Color
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+
+import com.google.zxing.BarcodeFormat
+import com.google.zxing.EncodeHintType
+import com.google.zxing.MultiFormatWriter
+import com.google.zxing.WriterException
+import com.google.zxing.common.BitMatrix
+import kotlinx.android.synthetic.main.fragment_profile.*
+
+import org.hackillinois.android.R
+import org.hackillinois.android.database.entity.Attendee
+import org.hackillinois.android.database.entity.QR
+import org.hackillinois.android.database.entity.User
+import org.hackillinois.android.viewmodel.ProfileViewModel
+
+import java.util.EnumMap
+
+class ProfileFragment : Fragment() {
+    private var qrImageView: ImageView? = null
+    private var nameTextView: TextView? = null
+    private var dietTextView: TextView? = null
+
+    private var universityLabel: TextView? = null
+    private var universityTextView: TextView? = null
+    private var majorLabel: TextView? = null
+    private var majorTextView: TextView? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val viewModel = ViewModelProviders.of(this).get(ProfileViewModel::class.java)
+        viewModel.init()
+
+        viewModel.qr.observe(this, Observer { qr -> updateQrView(qr) })
+        viewModel.user.observe(this, Observer { user -> updateUserInformation(user)})
+        viewModel.attendee.observe(this, Observer { attendee -> updateAttendeeInformation(attendee)})
+    }
+
+    private fun updateQrView(qr: QR?) = qr?.let {
+        val text = qr.qrInfo
+        val bitmap = generateQR(text)
+        qrImageView?.setImageBitmap(bitmap)
+    }
+
+    private fun updateUserInformation(user: User?) = user?.let {
+        nameTextView?.text = user.getFullName()
+    }
+
+    private fun updateAttendeeInformation(attendee: Attendee?) {
+        attendee?.let {
+            val diet = attendee.getDietAsString()
```

Comment:
I think I wrote this haha but can we always set the text, and then stick the VISIBLE/INVISIBLE decision in a ternary?

Reply:
Like line 74

Reply:
What do we set the text to if diet is null?